waspc/src/Wasp/Psl/Parser/ConfigBlock.hs

```diff
@@ -0,0 +1,59 @@
+module Wasp.Psl.Parser.ConfigBlock
+  ( configBlock,
+  )
+where
+
+import Text.Parsec
+  ( choice,
+    many,
+    newline,
+    optional,
+    try,
+  )
+import Text.Parsec.String (Parser)
+import qualified Wasp.Psl.Ast.ConfigBlock as Psl.ConfigBlock
+import Wasp.Psl.Parser.Argument (expression)
+import Wasp.Psl.Parser.Common
+  ( braces,
+    identifier,
+    reserved,
+    whiteSpace,
+  )
+
+-- | Parses a config block.
+-- Example of PSL config block:
+-- datasource db {
+--   provider   = "postgresql"
+--   url        = env("DATABASE_URL")
+--   extensions = [hstore(schema: "myHstoreSchema"), pg_trgm, postgis(version: "2.1")]
+-- }
+configBlock :: Parser Psl.ConfigBlock.ConfigBlock
+configBlock = do
+  whiteSpace
+  configBlockType <-
+    choice
+      [ try (reserved "datasource" >> return Psl.ConfigBlock.Datasource),
+        try (reserved "generator" >> return Psl.ConfigBlock.Generator)
+      ]
+  name <- identifier
+  content <- configBlockBody
+  optional newline
+  return $ Psl.ConfigBlock.ConfigBlock configBlockType name content
+
+configBlockBody :: Parser [Psl.ConfigBlock.KeyValuePair]
+configBlockBody = braces (many keyValue)
+
+-- | Parses a key-value pair.
+-- Example of PSL key-value pair:
+-- provider = "postgresql"
+-- It works for both datasources and generators.
+keyValue :: Parser Psl.ConfigBlock.KeyValuePair
+keyValue = do
+  whiteSpace
+  key <- identifier
+  whiteSpace
+  reserved "="
+  whiteSpace
+  value <- expression
+  optional whiteSpace
```

Comment:
I wonder why this one is optional, but the one at the start of the parser is not? Does that mean there has to be a whiteSpace at the start?

Reply:
Ok I just found this in the docs:
> Parses any white space. White space consists of zero or more occurrences of a [space](https://hackage.haskell.org/package/parsec-3.1.17.0/docs/Text-Parsec-Char.html#v:space), a line comment or a block (multi line) comment. Block comments may be nested. How comments are started and ended is defined in the [LanguageDef](https://hackage.haskell.org/package/parsec-3.1.17.0/docs/Text-Parsec-Token.html#t:LanguageDef) that is passed to [makeTokenParser](https://hackage.haskell.org/package/parsec-3.1.17.0/docs/Text-Parsec-Token.html#v:makeTokenParser).

So from this, sounds to me like `optional` here is redundant?

This also explains to me how we parse comments! That does mean we also drop them though hm. But ok. Btw I tihnk Prisma normally doesn't parse comments, they carry them over. If we are dropping them, and then using this same AST to generate Prisma code again, that means we are behaving different than Prisma's parser: we drop the comments, they don't. I think that is okish for now, but it is good to be aware of that.

Reply:
It is also interesting to consider if whiteSpace at start and end here should be really ahppening inside of this function, or should it instead be applied on the call site. I am not sure myself. In theory, doing it on the call site sounds better, because then you have the flexbility to also force no whitespace before or after keyvalue if you want that. But we will probably never need that so this is probably more practical. I am ok with this, just thinking out loud here.

Reply:
Actually, this got me wondering about whiteSpace, what is the right way to use it. Where it should be put and when.

Check out `lexeme` in this doc: https://hackage.haskell.org/package/parsec-3.1.17.0/docs/Text-Parsec-Token.html . I think this gives an idea.

So a lot of their parers, that come with Parsec, are already lexeme parsers and they take care of whiteSpace after themselves.

What we probably want to do is also use this `lexeme` thing? Not 100% sure. Maybe some googling around this would close the deal. I also don't want us to spend too much time on this, if we have something taht works. But this thing with whiteSpaces sounds like it is easy to make mistake if yo need to keep remaking this decision where to put it, so having some rule / unified approach around it might be valuable.

Reply:
TLDR: I don't want us to waste too much time, but take a look at lexeme, and see if it make sense for us to use sometihng like that through the code consistnely instead of whiteSpace directly. If yes, great, if not, ok. But in any case, let's consider removing those `optional`'s in front of whiteSpace, I think they are redundant.